corehq/apps/sso/models.py

```diff
@@ -276,3 +358,17 @@ class Meta:
 
     def __str__(self):
         return f"{self.domain} trusts [{self.identity_provider.name}]"
+
+
+@receiver(pre_save, sender=TrustedIdentityProvider)
+@receiver(pre_delete, sender=TrustedIdentityProvider)
+def clear_caches_when_trust_is_established_or_removed(sender, instance, **kwargs):
```

Comment:
Same race condition question as above.

Reply:
updated to `post-save` and `post-delete`